### PR TITLE
fixes #23759; rework move for refc

### LINKED ISSUE #23759: `move(table)` does not move the table
### Description

```nim
import tables
var a, b: Table[int, int]
a = move(b)
```

a copy is generated even if a move is requested:

```c
	genericSeqAssign((&a__testit_u10.data), b__testit_u11.data, (&NTIkeyvaluepairseq__E8pi9b5QNahsURYzXMjh3qw_));
	a__testit_u10.counter = b__testit_u11.counter;
	unsureAsgnRef((void**)&b__testit_u11.data, NIM_NIL);
	b__testit_u11.counter = 0;
```

orc moves.

### Nim Version

2.0.6/refc

### Current Output

_No response_

### Expected Output

_No response_

### Possible Solution

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ringabout**
fixes #23759

### CODE REVIEW

File: compiler/ccgexprs.nim [hunk 1]
```diff
@@ -2345,7 +2345,7 @@ proc genMove(p: BProc; n: PNode; d: var TLoc) =
           else:
             linefmt(p, cpsStmts, "$1($2);$n", [rdLoc(b), byRefLoc(p, a)])
     else:
-      let flags = if not canMove(p, n[1], d): {needToCopy} else: {}
+      let flags = if isSinkType(n[1].typ): {needToCopy} else: {}
```
Comment (Araq) [inline compiler/ccgexprs.nim]: Huh? You cannot move just because it is a sink type and also, why not patch `canMove` instead?

Comment (ringabout) [reply compiler/ccgexprs.nim]: In what situation that an object cannot be moved in refc? Before https://github.com/nim-lang/Nim/pull/22229, there is basically no restriction on `move` in refc.

**Comment by Araq** [comment]
This is not good enough, yet:

```nim

proc p(x: sink string) = 
  var y = move(x)
  assert x.len == 0
```

File: compiler/ccgexprs.nim [hunk 2]
```diff
@@ -2345,8 +2345,13 @@ proc genMove(p: BProc; n: PNode; d: var TLoc) =
...
-      let flags = if not canMove(p, n[1], d): {needToCopy} else: {}
-      genAssignment(p, d, a, flags)
+      if n[1].kind == nkSym and isSinkParam(n[1].sym):
+        var tmp = getTemp(p, n[1].typ.skipTypes({tySink}))
+        genAssignment(p, tmp, a, {needToCopySinkParam})
+        genAssignment(p, d, tmp, {})
+        resetLoc(p, tmp)
+      else:
+        genAssignment(p, d, a, {})
       resetLoc(p, a)
```
Comment (Araq) [inline compiler/ccgexprs.nim:2355]: If you always reset `a` then you mutate const data, potentially. What you need to do is to shadow the sink parameter with a variable of the same name.

**Pull request merged**